APE9.rst

```diff
@@ -77,6 +77,192 @@ This APE proposes that we place the following restrictions on binned data:
 #. The width of the bins is stored in a second column containing ``TimeDelta`` objects.
 
 
+Indexing
+--------
+
+Indexing by Time-Like Objects
+#############################
+
+When indexing using time-like values you get a TimeSeries returned.
+A single index value can be used:
+
+::
+	single_row_timeseries = timeseries[datetime]
+
+
+Which returns a Timeseries with a single row and the same columns as the original.
+
+A range of datetimes:
+
+::
+	multi_row_timeseries = timeseries[datetime_start: datetime_end]
+
+
+Which returns a TimeSeries truncated between the datetime_start (inclusive) and datetime_end (exclusive?).
+
+*Note: neither datetime_start nore datetime_end need to be exact index value in the original Timeseries.*
+
+If you input a time-like value not in the index then raise an IndexError.
+
+
+
+	
+Indexing by Integer (i-index)
+#############################
+
+Similar to time-like indexing either implicitly or using the .iloc attribute: ::
+	single_row_timeseries = timeseries[i_integer] = timeseries.iloc[i_integer]
+
+::
+	multi_row_timeseries = timeseries[i_integer_start: i_integer_end] = timeseries.iloc[i_integer_start: i_integer_end]
+  
+With support of negative index values:
+
+::
+	single_row_timeseries = timeseries[-i_integer] = timeseries.iloc[-i_integer] = timeseries.iloc[-i_integer] = timeseries.iloc[-i_integer]
+ 
+::
+	multi_row_timeseries = timeseries[i_integer_start: -i_integer_end] = timeseries.iloc[i_integer_start: -i_integer_end]
+	
+
+
+Indexing by Column Name (String)
+################################
+
+When indexing using a String for the column name you get a Timeseries with the given column/s.
+
+You can use a single column name:
+
+::
+	single_column_timeseries = timeseries[colname_string]
+  
+Which returns a Timeseries with a single column and the same rows/indices as the original.
+
+You can use a multiple column:
+
+::
+	multi_column_timeseries = timeseries[colname_string1, colname_string2, colname_string3]
+  
+Which returns a Timeseries with a multiple columns (in the given order) and the same rows/indices as the original.
+
+The names for the columns should be accessable, for example using the Table-like attribute colnames:
+
+::
+	timeseries.colnames
+
+*Note: Pandas DataFrames have a similar columns attribute, but AstroPy Table have an attribute with this name that returns a TableColumns object.*
+	
+
+Indexing by Bolean Array
+########################
+
+You can select multiple disjoint rows using a 1D array of Boolean values (of same length as the number of TS rows), a Timeseries is returned with all the rows corresponding to True, as with Numpy Array, Pandas DataFrame and AstroPy Table:
+
+::
+	multi_row_timeseries = timeseries[boolean_array]
+
+This is often used when filtering, for example based on value:
+
+::
+	multi_row_timeseries = timeseries[timeseries['col_b'] > 5.2]
```

Comment:
This leads to an interesting question, if `timeseries['col_b']` returns a timeseries how do you define the `>` operator on a timeseries object?